repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/openapivalidator/InvalidOpenApiDocumentTest.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package features.filters.openapivalidator
+
+import org.junit.experimental.categories.Category
+import org.openrepose.framework.test.ReposeValveTest
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.MessageChain
+import scaffold.category.Filters
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Verifies the behavior of the OpenAPI Validator Filter when loading an invalid OpenAPI document.
+ */
+@Category(Filters)
+class InvalidOpenApiDocumentTest extends ReposeValveTest {
+
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+    }
+
+    def "when configured with an invalid OpenAPI document, the filter should fail to initialize"() {
```

Comment:
Even though both of the configurations are invalid, as written it would only take one of them being invalid for the test to pass.

Reply:
Actually, this test does require two failures in that it searches for two logs line noting the failure. 

Really, my intent with this test was just to prove that an invalid OpenAPI document would lead to a startup error, and so I used a v2 and v3 document for good measure, but that may not have been strictly necessary in the first place.

I would really rather not have to start up Repose twice to get true isolation to check with a v2 and v3 document. So I could drop one of those documents from the test, but I think I prefer the way I've done it here.

Reply:
Fair enough.